.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,44 @@
+---
+name: CodeQL
+
+# Cancel previous workflow run groups that have not completed.
+concurrency:
+  # Group workflow runs by workflow name, along with the head branch ref of the pull request
+  # or otherwise the branch or tag ref.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+# Disable permissions for all available scopes.
+# Enable permissions for specific scopes as needed on job level.
+permissions: {}
+
+on:
+  push:
+    # Only run if JS files changed.
+    paths:
+      - '**.js'
```

Comment:
Does CodeQL only cover first-party JS? What about dependencies? Just curious, I'm not familiar with CodeQL.

I'm only asking since, if it also covers dependencies, we may also want to include `package.json` and `package-lock.json` here.

Reply:
@felixarntz Fpr dependencies, @dependabot creates security alerts that can be configured from -  https://github.com/WordPress/performance/security/dependabot